portal/src/main/webapp/js/src/oncoprint/webgl/setup.js

```diff
@@ -1162,45 +1277,76 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
 	return State.addAndPopulateClinicalTracks(attr);
     };
 
-    (function initOncoprint() {
+    (/**
+      * Initializes the OncoPrint tracks, populates them and scrolls back.
+      *
+      * @returns {Promise} - promise that gets fulfilled when the
+      * initialization process is complete, or rejected if data
+      * could not be retrieved.
+      */
+     function initOncoprint() {
 	LoadingBar.show();
 	LoadingBar.msg(LoadingBar.DOWNLOADING_MSG);
-	var def = new $.Deferred();
 	oncoprint.setCellPaddingOn(State.cell_padding_on);
-	$.when(QuerySession.getOncoprintSampleGenomicEventData()).then(function (oncoprint_data) {
+	console.log("in initOncoprint, fetching genomic event data");
+	return QuerySession.getOncoprintSampleGenomicEventData()
+	.then(function (oncoprint_data) {
+	    console.log("in initOncoprint, adding genomic event tracks");
 	    State.addGeneticTracks(oncoprint_data);
-	}).fail(function () {
-	    def.reject();
-	}).then(function () {
-	    (function fetchClinicalAttributes() {
-		QuerySession.getClinicalAttributes().then(function (attrs) {
-		    State.unused_clinical_attributes = attrs;
-		    State.clinical_attributes_fetched.resolve();
-		    def.resolve();
-		}).fail(function () {
-		    def.reject();
-		    State.clinical_attributes_fetched.reject();
+	    // return the promise needed for the next step
+	    return QuerySession.getDefaultHeatmapProfile();
+	}).then(function (heatmap_profile_id) {
+	    if (heatmap_profile_id !== null) {
+		console.log("in initOncoprint, fetching heatmap data");
+		return QuerySession.getHeatmapData(heatmap_profile_id, QuerySession.getQueryGenes(), "sample")
+		.then(function (heatmap_data) {
+		    console.log("in initOncoprint, heatmap data fetched, adding tracks");
+		    State.addHeatmapTracks(heatmap_data);
 		});
-	    })();
-	})
-	return def.promise();
-    })().then(function () {
-        var populate_data_promise = State.setDataType(State.using_sample_data ? 'sample' : 'patient');
-	    
-        $.when(QuerySession.getPatientIds(), QuerySession.getAlteredSamples(), QuerySession.getAlteredPatients(), QuerySession.getCaseUIDMap(), populate_data_promise).then(function(patient_ids, altered_samples, altered_patients, case_uid_map) {
-	    if ((State.using_sample_data ? window.QuerySession.getSampleIds() : patient_ids).length > 200) {
-		// TODO: assume multiple studies
-		var study_id = QuerySession.getCancerStudyIds()[0];
-		var getUID = function(id) {
-		    return case_uid_map[study_id][id];
-		};
-		oncoprint.setHorzZoomToFit(State.using_sample_data ? altered_samples.map(getUID) : altered_patients.map(getUID));
+	    } else {
+		// heatmap not applicable, return a resolved Promise
+		return $.when();
 	    }
-	    oncoprint.scrollTo(0);
-	});
-	
-	return populate_data_promise;
-    });
+	}).then(function fetchClinicalAttributes() {
+	    console.log("in initOncoprint, fetching clinical attributes");
```

Comment:
I addressed this in https://github.com/cBioPortal/cbioportal/pull/1847/commits/86ece36645fdaa1f43266ea7c5e228788ac344a7